### ISSUE TITLE
Activation function for object proposals in RoI (test time)

### ISSUE THREAD

**Issue by AlexTS1980** [pytorch/vision#2894]
## 🚀 Feature
Replace softmax with sigmoid in `postprocess_detections `method in `roi_heads`:
https://github.com/pytorch/vision/blob/5cb77a20c3c65ca6199fdf1c1bc642af7447d311/torchvision/models/detection/roi_heads.py#L677         

## Motivation
In the current implementation, score is class-dependent (softmax), but NMS is class-independent. So the question is, can/should one RoI output more than 1 prediction.  
## Pitch

If there are` C` classes, each RoI outputs `C` score and bounding box predictions (two tensors, size` (1, C)` and `(4,C)` resp.) at test stage (`postprocess_detections `method). Non-max suppression is done independently of the class (i.e. boxes overlapping more than NMS are kept if they are different classes). But the normalization function is not class-independent:

`pred_scores = F.softmax(class_logits, -1)`

So if there are two positive classes, pred_scores vector will be, e.g. [0.9, 0.1], and at some point both of these scores will be compared to `box_score_thresh`. Obviously one of them is very likely to be rejected. Therefore, I don’t quite understand this implementation. It should be either:

```
pred_scores = F.sigmoid(class_logits, -1)
preds =  torch.nonzero(pred_scores.sigmoid()>box_score_thresh)
```

to compute the scores independently, or

```
preds = class_logits.max(-1)
preds.values[preds.indices>0].sigmoid()>box_score_thresh
``` 

to extract the best prediction from every RoI. Then the predictions will be independent. I think it needs to be re-implemented or at least added as an argument to choose from. Mask predictions are done independently in this way:

https://github.com/pytorch/vision/blob/5cb77a20c3c65ca6199fdf1c1bc642af7447d311/torchvision/models/detection/roi_heads.py#L73

**Comment by fmassa** [comment]
Hi,

The current implementation follows what Faster R-CNN originally does, and as such we will be this as is in torchvision for consistency.

The R-CNN family of models is trained such that each region has only one label to it. As such, if two objects are highly overlapping, Faster R-CNN by construction is not able to handle them properly.

The current scheme for NMS is inherited from R-CNN (and maybe even before with DPM), and is a way of improving recall (and thus mAP).

The mask head is different, as there is only one single class (foreground), and thus a sigmoid is adapted.

I recommend watching [Object Detection as a Machine Learning Problem by Ross Girshick](https://www.youtube.com/watch?v=her4_rzx09o) that explains a bit the heuristics for training / evaluating detection models traditionally, and I would also recommend you looking at [our paper DETR](https://arxiv.org/abs/2005.12872) or the [video explaining it](https://www.youtube.com/watch?v=utxbUlo9CyY), which does not need those heuristics during training / testing, and is by construction able to detect overlapping objects.
